code/modules/power/supermatter/supermatter.dm

```diff
@@ -241,11 +302,71 @@
 		var/rads = (power / 10) * sqrt( 1 / max(get_dist(l, src),1) )
 		l.rad_act(rads)
 
-	power -= (power/500)**3
+	if(power > POWER_PENALTY_THRESHOLD)
+		playsound(src.loc, 'sound/weapons/emitter2.ogg', 100, 1, extrarange = 10)
+		supermatter_zap(src, 5, min(power*2, 20000))
+		supermatter_zap(src, 5, min(power*2, 20000))
+		if(power > SEVERE_POWER_PENALTY_THRESHOLD)
+			supermatter_zap(src, 5, min(power*2, 20000))
+			if(power > CRITICAL_POWER_PENALTY_THRESHOLD)
+				supermatter_zap(src, 5, min(power*2, 20000))
 
-	return 1
+		if(prob(15))
+			supermatter_pull(src, power/750)
+		if(prob(5))
+			supermatter_anomaly_gen(src, 1, rand(5, 10))
+		if(power > SEVERE_POWER_PENALTY_THRESHOLD && prob(5) || prob(2))
+			supermatter_anomaly_gen(src, 2, rand(5, 10))
+		if(power > SEVERE_POWER_PENALTY_THRESHOLD && prob(2) || prob(0.3))
+			supermatter_anomaly_gen(src, 3, rand(5, 10))
 
-/obj/machinery/power/supermatter_shard
+
+
+	if(damage > warning_point) // while the core is still damaged and it's still worth noting its status
+		if((world.timeofday - lastwarning) / 10 >= WARNING_DELAY)
```

Comment:
err why are you using timeofday? just use world.time and count in ticks instead.

time of day is susceptible to issues caused when it resets at midnight